internal/lm/nvml.go

```diff
@@ -218,6 +233,90 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(configFile string, device []resource.Device) (Labeler, error) {
+	if configFile == "" {
+		return nil, nil
+	}
+
+	// Read file and parse it
+	imexConfig, err := os.Open(configFile)
+	if err != nil && os.IsNotExist(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to open imex config file: %v", err)
+	}
+	defer imexConfig.Close()
+
+	// Read the file line by line
+	var ips []string
+	scanner := bufio.NewScanner(imexConfig)
+	for scanner.Scan() {
+		ip := strings.TrimSpace(scanner.Text())
+		if net.ParseIP(ip) == nil {
+			return nil, fmt.Errorf("invalid IP address in imex config file: %s", ip)
+		}
+		ips = append(ips, ip)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("failed to read imex config file: %v", err)
+	}
+
+	if len(ips) == 0 {
+		return nil, nil
+	}
+
+	// Sort the IP addresses
+	sort.Strings(ips)
+
+	// Join the sorted IPs into a single string
+	sortedIPs := strings.Join(ips, "\n")
+
+	var commonClusterUUID string
+	var commonCliqueID string
+	for _, d := range device {
+		ok, err := d.IsFabricAttached()
+		if !ok {
+			continue
+		}
+		if err != nil {
+			return nil, fmt.Errorf("error checking imex capability: %v", err)
+		}
+
+		clusterUUID, cliqueID, err := d.GetFabricIds()
+		if err != nil {
+			return nil, fmt.Errorf("error getting cluster UUID: %v", err)
+		}
+		if commonClusterUUID == "" {
+			commonClusterUUID = clusterUUID
+		}
+		if commonClusterUUID != clusterUUID {
+			klog.Warningf("Cluster UUIDs are different: %s != %s", commonClusterUUID, clusterUUID)
+			return nil, nil
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("error getting clique ID: %v", err)
+		}
+		if commonCliqueID == "" {
+			commonCliqueID = cliqueID
+		}
+		if commonCliqueID != cliqueID {
+			klog.Warningf("Clique IDs are different: %s != %s", commonCliqueID, cliqueID)
+			return nil, nil
+		}
+	}
+
+	labels := Labels{
+		"nvidia.com/gpu.clusteruuid": commonClusterUUID,
+		"nvidia.com/gpu.cliqueid":    commonCliqueID,
+		"nvidia.com/gpu.imex-domain": generateUUID(sortedIPs) + "-" + commonCliqueID,
```

Comment:
Does this make sense if `generateUUID` doesn't return the same value across restarts?

Reply:
given that is the same seed, the resulting UUID should be the same

Reply:
from all my testing, I have always got the same label `0f884867-ba2f-4294-9155-b495ff367eea` , so the UUID is tied to the Ip's , even after a GFD restart

Reply:
I wrote the generateUUID function this was borrowed from specifically to ensure that it always generates the same number given the same seed value.